### PR TITLE
Add service to create usage_charge_group when add a plan to customer

### PR DESCRIPTION
**Pull request by TronTram**
## What
- Added a new service `UsageChargeGroups::CreateService` for generating appropriate `usage charge groups` based on a subscription's plan upon subscription creation.

- Add a **unique index** to usage_charge_groups table on the combination of **charge_group_id** and **subscription_id**.

## Why
- To ensure that the system accurately creates usage charge group.

## Screenshot
- Create success:
image

### CODE REVIEW

File: db/migrate/20240311122947_add_unique_index_to_usage_charge_group.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class AddUniqueIndexToUsageChargeGroup < ActiveRecord::Migration[7.0]
```
Comment (TronTram) [inline db/migrate/20240311122947_add_unique_index_to_usage_charge_group.rb:1]: Style/FrozenStringLiteralComment: Missing frozen string literal comment.

File: app/services/usage_charge_groups/create_service.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  class CreateService < BaseService
+    def initialize(subscription:)
+      super
+      @subscription = subscription
+    end
+
+    def call
+      is_group_charge?
+      create_usage_charge_group
+    end
```
Comment (tnlong311) [inline app/services/usage_charge_groups/create_service.rb:13]: What does `is_group_charge?` do? Do you mean: 
```ruby
creat_usage_charge_group if is_group_charge?
```

File: app/services/usage_charge_groups/create_service.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+    private
+
+    attr_reader :subscription
+
+    delegate :plan, to: :subscription
+
+    def is_group_charge?
```
[the remark this answers is not in the document]
Comment (tnlong311) [reply app/services/usage_charge_groups/create_service.rb:21]: or `has_any_group_charge`

File: app/services/usage_charge_groups/create_service.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+    def compute_available_group_usage(charge_group)
+      available_group_usage = {}
+      charge_group.charges.package_group.each do |charge|
+        billable_metric = charge.billable_metric
+        available_group_usage[billable_metric.id] = charge.properties['package_size']
+      end
+
+      available_group_usage
+    end
```
Comment (tnlong311) [inline app/services/usage_charge_groups/create_service.rb:45]: Should not create `available_group_usage` in a `usage_charge_group` initially. It must be null by default in order for the code logic to be correct. Thus, we can remove this function.

File: app/services/usage_charge_groups/create_service.rb (hunk 2 above)
Comment (TronTram) [reply app/services/usage_charge_groups/create_service.rb:13]: thank you

File: app/services/usage_charge_groups/create_service.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (TronTram) [reply app/services/usage_charge_groups/create_service.rb:21]: aggree! `has_group_charge` is better

File: app/services/usage_charge_groups/create_service.rb (hunk 4 above)
Comment (TronTram) [reply app/services/usage_charge_groups/create_service.rb:45]: usage_charge_group has validates :available_group_usage, presence: true. We need to remove the validation if we want to allow it to be null.

**Review by tnlong311 — APPROVED** [review]
LGTM

**Pull request merged**